lib/ansible/plugins/inventory/vmware_vm_inventory.py

```diff
@@ -132,6 +164,19 @@ def do_login(self):
         if self.with_tags:
             self.rest_content = self._login_vapi()
 
+        # Get the folder object that will serve as our inventory root folder
+        if self.search_folder:
+            self.root_folder = self.content.searchIndex.FindByInventoryPath('%s/vm/%s' % (self.datacenter, self.search_folder))
+            if not self.root_folder:
+                raise AnsibleError("Specified datacenter + search_folder '%s/vm/%s' were not found on Inventory." %
+                                   (self.datacenter, self.search_folder))
+        elif self.with_folders:
+            self.root_folder = self.content.searchIndex.FindByInventoryPath('%s/vm' % self.datacenter)
+            if not self.root_folder:
+                raise AnsibleError("Specified datacenter '%s' was not found on Inventory." % self.datacenter)
+        else:
+            self.root_folder = self.content.rootFolder
```

Comment:
I am thinking about - 
```suggestion
          elif self.datacenter:
            # User wants specific datacenter's VM
            datacenter_obj = self.content.searchIndex.FindByInventoryPath('%s' % self.datacenter)
            if datacenter_obj:
               self.root_folder = self.datacenter_obj.vmFolder
            else:
               raise AnsibleError("Unable to find the datacenter specified using %s" % self.datacenter)
          else:
            # User has not specified datacenter nor search_folder, with_folders
            self.root_folder = self.content.rootFolder
```

Reply:
That would be a nice addition. Then the datacenter param wouldn't be ignored anymore on absence of the folder params (requiring changes in the doc and the related test). I'll try it out to make sure it works as expected.

Reply:
Actually, I've just realised there's a shorter way. See my comment on line 179 (`elif self.with_folders`)